packages/launcher/src/components/launch/Launch.tsx

```diff
@@ -73,35 +79,118 @@ export default function Launch(): ReactElement {
   const [axonVersions, setAxonVersions] = React.useState([""]);
   const [requested, setRequested] = React.useState(false);
   const [axonVersion, setAxonVersion] = React.useState("edge");
+  const [internetConnection, setInternetConnection] = React.useState(false);
+  const [checkedInternetConnection, setCheckedInternetConnection] = React.useState(false);
+  const [hasImages, setHasImages] = React.useState(true);
+  const [clickedImageDialog, setClickedImageDialog] = React.useState(false);
 
   const getVersions = async () => {
     const ipcRenderer: IpcRenderer = window.require("electron").ipcRenderer;
 
-    ipcRenderer.on("axon-tags", (event, args: string[]) => {
+    await ipcRenderer.on("axon-tags", (event, args: string[]) => {
       setAxonVersions(args);
       setAxonVersion(args[0]);
     });
     ipcRenderer.send("request-tags");
     ipcRenderer.send("request-version");
   };
+  const getTagsFromSystem = async () => {
+    const connected = await docker.isConnected();
+    if (connected) {
+      const images = await docker.getImages();
+      if (images !== null && images.length > 0) {
+        const imageMap = new Map();
+        for (let i = 0; i < images.length; i++) {
+          const tmpTag = images[i].RepoTags[0].split(":")[1];
+          if (imageMap.has(tmpTag)) {
+            imageMap.set(tmpTag, imageMap.get(tmpTag) + 1);
+          } else {
+            imageMap.set(tmpTag, 1);
+          }
+        }
+        const tmpTags: string[] = [];
+        imageMap.forEach((value: number, key: string) => {
+          if (value >= 6) tmpTags.push(key);
+        });
+        console.log(tmpTags);
+        setAxonVersions(tmpTags);
+        setAxonVersion(tmpTags[0]);
+      }
+    }
+  };
+  const getInternetConnection = async () => {
+    const ipcRenderer: IpcRenderer = window.require("electron").ipcRenderer;
+    await ipcRenderer.on("internet-status", (event, arg: boolean) => {
+      setInternetConnection(arg);
+      setCheckedInternetConnection(true);
+      if (!arg) {
+        // need to look for own tags
+        getTagsFromSystem();
+      }
+    });
+    ipcRenderer.send("request-internet");
+  };
```

Comment:
I think we want to check if we are connected every so often. If we do not check, then users will need to restart the app: https://reactjs.org/docs/rendering-elements.html#updating-the-rendered-element